app-emulation/qemu/qemu-9999.ebuild

```diff
@@ -438,6 +434,14 @@ check_targets() {
 	popd >/dev/null
 }
 
+if [[ ${PV} == 9999 ]]; then
+src_unpack() {
+	git-r3_src_unpack
+	cd "${P}" || die
+	meson subprojects download keycodemapdb berkeley-softfloat-3 berkeley-testfloat-3 || die
```

Comment:
> I am not sure if this is correct approach (almost certainly not because it forfeits "download sources upfront" apporach), but I'm failing to come up with better.

Well, I'm not sure why it is so important to qemu that these be git repositories especially considering they are just copies of the 2018 stable release of http://www.jhauser.us/arithmetic/index.html with dos2unix run on a bunch of .c / .h files and I guess an additional source/RISCV/ directory that qemu isn't even using -- the canonical upstream location is anyways https://github.com/ucb-bar/berkeley-softfloat-3/ etc.

So IMO, qemu should be using tarball wraps, not git clone wraps. Then it would be trivial to manage -- just list the tarballs as additional distfiles and copy them into subprojects/packagecache/.

You can still clone the git repos yourself and copy them into the source tree. Instead of running `meson subprojects download`, use `meson subprojects packagefiles --apply`.

Reply:
Doesn't that require changing the qemu upstream first? I can try and post a patch for qemu, sure.

Reply:
... no?

I said if the qemu upstream was fixed then you'd just need to download tarball distfiles and cp them over and meson autodetects them.

"You can still copy git repos over but you'll need to run packagefiles --apply."

Reply:
Something like this?

https://github.com/gentoo/gentoo/pull/32684